test/cpp/c10d/ProcessGroupNCCLErrorsTest.cpp

```diff
@@ -286,7 +286,6 @@ TEST_F(ProcessGroupNCCLErrorsTest, testNCCLErrorsBlocking) {
 
   auto work = pg.allreduce(tensors_);
   work->wait();
-  EXPECT_TRUE(work->isSuccess());
```

Comment:
This is bc single device style is blocking and multi was non blocking ? Or what

Reply:
No, it is just because the `is_success` API has been deprecated for a long time at the pybind place:
https://github.com/pytorch/pytorch/blob/main/torch/csrc/distributed/c10d/init.cpp#L2556-L2560
so taking this chance to take it out as well.
Also because its c++ code would block my refactorization.

It makes sense (to deprecate it) -- how can we tell a collective is successful? Are we telling user that the result is correct? We obviously cannot do data check.. So at best we can only tell `is_complete()`.